src/components/AngerNoodle/helperFunctions/eventHandlers.js

```diff
@@ -0,0 +1,67 @@
+import { updateBoardData, copySnake, angryMessages } from "./data";
+
+const handleKeyPress = (
+  event, directionRef, gameStateRef, defaultSnake, updateSnake, updateBoard, updateScore, snakeRef, updateAngryMessage
+) => {
+  const direction = directionRef;
+  const gameState = gameStateRef;
+  const currentSnakeCopy = copySnake(snakeRef.current);
+  const restartGame = () => {
+    gameState.current = 'active';
+    direction.current = 'right';
+    updateAngryMessage(angryMessages.setDefault());
+    updateScore(0);
+    updateBoard(updateBoardData(defaultSnake, direction.current, [10, 5]));
+    updateSnake(defaultSnake);
+  }
+  switch (event.key) {
```

Comment:
You can refactor this using the logic I sent along the other day to cut out ~80% of it, I'd recommend doing that sooner rather than later so that you don't end up running into any bugs.

Alternatively, since you have a lot of repeated logic I'd look into breaking this into multiple functions